subsys/bluetooth/controller/hci_internal.c

```diff
@@ -1319,21 +1438,21 @@ int hci_internal_cmd_put(uint8_t *cmd_in)
 	} else if (!is_host_using_legacy_and_extended_commands(opcode)) {
 		cmd_put(cmd_in, &cmd_complete_or_status.raw_event[0]);
 	} else {
+#if !defined(CONFIG_BT_HCI_HOST)
 		/* The host is violating the specification
 		 * by mixing legacy and extended commands.
+		 * For a combined host and controller build, we know that the zephyr
+		 * host is used. For this case we know that the host is not
+		 * combining legacy and extended commands. Therefore we will never end
+		 * up in this case.
 		 */
 		if (command_generates_command_complete_event(opcode)) {
-			uint8_t param_length = sizeof(struct bt_hci_evt_cmd_complete)
-					     + sizeof(struct bt_hci_evt_cc_status);
-			(void)encode_command_complete_header(cmd_complete_or_status.raw_event,
-							     opcode,
-							     param_length,
-							     BT_HCI_ERR_CMD_DISALLOWED);
+			generate_status_disallowed_command_complete_event(cmd_in, opcode);
```

Comment:
I might be missing something, but the opcodes in `command_generates_command_complete_event ` are less than the ones in the newly added function `generate_status_disallowed_command_complete_event `
Shouldn't they be the same?